test/ranking.spec.js

```diff
@@ -0,0 +1,112 @@
+import {
+    Ranking
+} from '../src/app/Ranking';
+
+describe("Ranking's logic", () => {
+    const peopleRanking = new Ranking('people');
+
+    const results = [{
+        'user': 'user1',
+        'score': 15,
+        'maxScore': 30 // 50%
+    }, {
+        'user': 'user2',
+        'score': 18,
+        'maxScore': 40 // 45%
+    }, {
+        'user': 'user3',
+        'score': 15,
+        'maxScore': 25 // 60%
+    }];
+
+    afterEach(() => {
+        localStorage.clear();
+    });
+
+    it('When set wrong category name, exception will be thrown', () => {
```

Comment:
Super testy pokazują działanie klasy :) 

Choć do tej może bym się przyczepił. Czy to jest nam na pewno potrzebne? 
W przypadku dodania (załóżmy) nowego trybu gry to musimy tutaj zmienić tę listę. 
Ja bym nie uznał, że to jest błąd - ale sam programując bym tak nie zrobił.
Zależy co chcemy osiągnąć. W TypeScript pewnie mielibyśmy tutaj też typ od razu ograniczający możliwe wartości.

Reply:
Chodzi o tę listę w konstruktorze? Celowo dodałam walidację, aby przez chociażby literówkę, nie został stworzony osobny ranking/wynik zapisany w niewłaściwym miejscu. Choć może będzie to niedopuszczalne przez implementację mechanizmu, który będzie te wyniki zbierał i przekazywał do zapisania i taki ewentualny błąd wykluczy, ale że teraz ten mechanizm jeszcze nie istnieje, nie wiem jak będzie wyglądał, to chciałam zrobić takie zabezpieczenie, choć może faktycznie zbędne.

Reply:
Mhm. Też mi się wydaje, że to powinno być gdzieś wyżej. Ale to może zostawmy - najwyżej ktoś robiący warstawy używające przeniesie to.